### ISSUE TITLE
Basic test case: monorepos, SRPM builds pending

### ISSUE THREAD

**Issue by jpopelka** [packit/packit-service#2122]
All SRPM builds submitted from packit/hello-world#1557 are marked as `pending` in [(staging) dashboard](https://dashboard.stg.packit.dev/jobs/srpm-builds) even they succeeded.

![Screenshot from 2023-07-19 12-34-37](https://github.com/packit/packit-service/assets/288686/cf495f3d-f080-4ce3-9502-312021fc06f6)

**Comment by mfocko** [comment]
Wouldn't be surprised if it's related to: https://github.com/packit/packit-service/issues/2120

will check, though I'm not sure why it's not being picked up by the babysitting

**Comment by jpopelka** [comment]
I don't see any occurrence of #2120 on the staging

**Comment by lbarcziova** [comment]
Thanks @majamassarini for the investigation! I was not aware of this problem when implementing that check. In that case, I would probably go with storing the package name in the DB for the particular build (both SRPM and Copr) and checking that one in the pre-check method. WDYT?

**Comment by majamassarini** [comment]
If COPR server does not tell us how can we guess which `copr build id` is for which `package`?
I don't think we can store anything in the db, but maybe I am wrong.
On the other hand, if we don't want to ask a change in COPR, we can serialize the SRPM builds for the monorepo packages and skip the check. If we serialize the builds we are sure that the last `srpm-builds` event we receive is for the last submitted build.

**Comment by lbarcziova** [comment]
> If COPR server does not tell us how can we guess which copr build id is for which package?

Not sure I follow, there should be a separate Copr build (ID) for each package, shouldn't there?

**Comment by majamassarini** [comment]
Yes, and we don't know which event (and the related copr build id) is for which package.

**Comment by lbarcziova** [comment]
But if we store the package name with the build in our DB when submitting it, we can then later check it when processing the status of the build.

**Comment by majamassarini** [comment]
I see, I thought the call was async, but it isn't. So you are right we probably have the build id when submitting it and we can store there the package name 👍🏻

**Comment by majamassarini** [comment]
> I believe no change is needed in Copr, if I follow you correctly -> otherwise please fill an issue against https://github.com/fedora-copr/copr

Thanks Pavel, no changes in COPR should be needed. We will store the relevant data in our database.
